### PR TITLE
check-github-usernames: add script to check if all GitHub usernames are valid

### PR DESCRIPTION
**Pull request by sebastiaanspeck**
Inspired on #12506, but now to verify if all GitHub usernames in CODEOWNERS + MAINTAINERS.md are valid

### CODE REVIEW

File: .github/workflows/monthly-check.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Monthly check
```
Comment (spageektti) [inline .github/workflows/monthly-check.yml:1]:
```suggestion
name: Monthly check GitHub usernames
```
For me, the name "Monthly check" is too generic and can be misleading.

**Review by spageektti — APPROVED** [review]

**Comment by spageektti** [comment]
Any updates?

**Review by sbrl — COMMENTED** [review]
This in general looks good! It sounds like a useful check. What kind of actions would you suggest could be taken based on the data gathered? Removing contributors without an account does not sound like a good idea to me, because even though they may have later deleted their account that doesn't mean to say their contributions were not valuable.

**Comment by spageektti** [comment]
>  My suggestion for the MAINTAINERS.md is maybe to add a little note that their account has been removed? This script could ignore those

Also if the user was marked as a present maintainer, his entry in MAINTAINERS.md should be moved to the past maintainers' section.

**Pull request closed without merging**